### PR TITLE
docs(sass): add unified sass reference

### LINKED ISSUE #407: docs(core): document all public Sass variables and mixins
docs(core): document all public scss variables picking up the ones we think are required for customisation or extending.

### PR DESCRIPTION
**Pull request by rbarilani**
Use sassdoc to write, parse documentation and use that data to generate
documentation content visible in the website.

* remove sass reference for structure-and-grid section and core section
* add demo:sassdoc gulp task
* hook the new task into build and watch tasks
* add new templates for displaying the sass reference
* replace toolkit.yaml with autogenerated sassdoc_colors.json

close #407

### CODE REVIEW

**Comment by rbarilani** [comment]
👍

**Comment by rbarilani** [comment]
👍

File: src/styles/core/_variables.scss [hunk 1]
```diff
@@ -203,24 +117,38 @@ $dc-z-index100: 9999 !default;
...
+
+/// medium screens breakpoint 600px {Tablets}
+/// @type Number
 $dc-medium-width: 37.5em !default;
+
+/// large screens breakpoint 1024px {Desktops}
+/// @type Number
 $dc-large-width: 64em !default;
+
+/// huge screeens breakpoint 1280px {Full HD}
```
Comment (sambhav-gore) [inline src/styles/core/_variables.scss]: there is a typo here for 'screens'

**Comment by rbarilani** [comment]
@sambhav-gore fixed, and added a minimal description as you requested. please have a look

**Comment by rbarilani** [comment]
👍

**Comment by sambhav-gore** [comment]
👍

**Pull request merged**